tsdb/chunkenc/chunk.go

```diff
@@ -82,8 +82,20 @@ type Chunk interface {
 // Appender adds sample pairs to a chunk.
 type Appender interface {
 	Append(int64, float64)
-	AppendHistogram(t int64, h *histogram.Histogram)
-	AppendFloatHistogram(t int64, h *histogram.FloatHistogram)
+
+	// Appends a histogram sample to the Chunk.
+	// prev Appender is used to determine counter reset, only set when starting new empty chunk.
+	// Returned Chunk is nil if sample could be appended to current Chunk.
+	// If sample cannot be appeneded, a new Chunk is returned which is either the current Chunk recoded or a completely new Chunk.
+	// The Appender to use next is always returned.
+	AppendOrCreateHistogram(prev *HistogramAppender, t int64, h *histogram.Histogram, appendOny bool) (c Chunk, isRecoded bool, app Appender, err error)
```

Comment:
I wonder if this is the right interface if we need complete transparency when encoding. From a usability perspective, I would be happiest if I would create an appender, append histograms to it, and at the end get back a list of chunks. Internally the appender would keep track of when to recode, start a new chunk, and how to set the counter reset hint.

Does that seems like something plausible to implement?

Reply:
I agree that this is a bit convoluted to use still and that would be an ideal goal. I'd need to write down and compare the three use cases it's used for to figure out a better solution. Question is if this is a good enough first step, or we want to make it "clean" right away?

Reply:
I think it is fine to move forward with this approach, we can get there iteratively 👍